linode/nodepool/framework_resource_schema.go

```diff
@@ -0,0 +1,74 @@
+package nodepool
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+var resourceSchema = schema.Schema{
+	Version: 1,
+	Attributes: map[string]schema.Attribute{
+		"id": schema.Int64Attribute{
+			Description: "ID of the Node Pool.",
+			Computed:    true,
+		},
```

Comment:
Because of an ongoing issue with a downstream consumer (Pulumi), we'll need to treat ID attributes as `StringAttribute`s for the time being

Reply:
done